### PR TITLE
Fix maybe-uninitialised sequence number

### PR DESCRIPTION
**Pull request by benjaminbollen**

### CODE REVIEW

**Comment by benjaminbollen** [comment]
Hi Peter, yes I'm scratching my head over it as well.  Asked Fraser as well, but seemed best to ask your advice again.  Seems not a trivial bug that I can fix out of the box.

File: include/maidsafe/crux/detail/sequence_number.hpp [hunk 1]
```diff
@@ -36,6 +36,7 @@ class sequence_number {
 public:
     sequence_number();
     sequence_number(const sequence_number&);
+    sequence_number(sequence_number&&);
```
Comment (breese) [inline include/maidsafe/crux/detail/sequence_number.hpp]: Is this necessary? There is no implementation for it.

If it is necessary, then I think that we need to add the move assignment operator as well to observe the rule of five.

File: include/maidsafe/crux/detail/sequence_number.hpp [hunk 2]
```diff
@@ -77,8 +78,8 @@ sequence_number<NumericType, Max>::sequence_number(const sequence_number& other)
 }
 
 template<typename NumericType, NumericType Max>
-sequence_number<NumericType, Max>::sequence_number(NumericType n)
-    : n(n)
+sequence_number<NumericType, Max>::sequence_number(NumericType other_n)
+    : n(other_n)
```
Comment (breese) [inline include/maidsafe/crux/detail/sequence_number.hpp]: Am I correct to assume that this has nothing to do with the warning? The original code is both valid and correct.

Comment (inetic) [inline include/maidsafe/crux/detail/sequence_number.hpp]: Yes, I think the warning on the build machines has not been solved by these changes. I was being happy too soon in my previous comment

Comment (benjaminbollen) [inline include/maidsafe/crux/detail/sequence_number.hpp]: Yes Bjorn and Peter, these code changes are not related to the warnings.  Mere attempts to get GCC to be silent without silencing it.  To no avail.
